sdk/digitaltwins/azure-digitaltwins/azure/digitaltwins/_digitaltwin_models_client.py

```diff
@@ -0,0 +1,116 @@
+        :**kwargs The operation options
+        :type **kwargs: any
+        :returns: The ModelDate object
+        :rtype: ~azure.digitaltwins.models.ModelData
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        return self._client.digital_twin_models.get_by_id(model_id, include_model_definition, **kwargs)
+
+    def list_models(self, dependencies_for, include_model_definition=False, max_item_count=-1, **kwargs):
+        # type: (str, bool, int, **Any) -> Iterable["models.PagedModelDataCollection"]
+        """Get the list of models
+
+        :param dependencies_for: The model Ids to have dependencies retrieved.
+        If omitted, all models are retrieved
+        :type dependencies_for: str
+        :param include_model_definition: When true the model definition
+        will be returned as part of the result
+        :type include_model_definition: bool
+        :param max_item_count: The maximum number of items to retrieve per request.
+        The server may choose to return less than the requested max.
+        :type max_item_count: int
+        :**kwargs The operation options
+        :type **kwargs: any
+        :returns: An iterator like instance of either PagedModelDataCollection
+        :rtype: ~azure.core.paging.ItemPaged[~azure.digitaltwins.models.PagedModelDataCollection]
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        digital_twin_models_list_options = {'max_item_count': -1}
+        if max_item_count != -1:
+            digital_twin_models_list_options= {'max_item_count': max_item_count}
+        return self._client.digital_twin_models.list(
+            dependencies_for=dependencies_for,
+            include_model_definition=include_model_definition,
+            digital_twin_models_list_options=digital_twin_models_list_options, **kwargs
+        )
+
+    def create_models(self, models=None, **kwargs):
+        # type: (List[object], **Any) -> List["models.ModelData"]
+        """Create one or more models. When any error occurs, no models are uploaded.
+
+        :param models: The set of models to create. Each string corresponds to exactly one model.
+        :type models: List[object]
+        :type **kwargs: any
+        :returns: The list of ModelData
+        :rtype: List[~azure.digitaltwins.models.ModelData]
+        :raises :class: `~azure.core.exceptions.HttpResponseError`
+        """
+        return self._client.digital_twin_models.add(models, **kwargs)
+
+    def decommission_model(self, model_id, model_patch, **kwargs):
+        # type: (str, List[object], **Any) -> List["models.ModelData"]
+        """Updates the metadata for a model
+
+        :param model_id: The id for the model. The id is globally unique and case sensitive.
+        :type model_id: str
+        :param update_model: An update specification described by JSON Patch. Only the decommissioned
+        property can be replaced.
```

Comment:
What do I replace the decomissioned property to be? It seems like this could be done internally?

Reply:
Good catch, we won't even expose the "model_patch" here. The purpose of this function to decommission the model using the pre-defined patch. I updated it accordingly.

---
In reply to: [491745617](https://github.com/Azure/azure-sdk-for-python/pull/13852#discussion_r491745617) [](ancestors = 491745617)